uni/lib/view/login/login.dart

```diff
@@ -37,10 +41,67 @@ class LoginPageViewState extends State<LoginPageView> {
   bool _keepSignedIn = true;
   bool _obscurePasswordInput = true;
   bool _loggingIn = false;
+  bool _intercepting = false;
 
-  Future<void> _login(BuildContext context) async {
-    final stateProviders = StateProviders.fromContext(context);
-    final sessionProvider = stateProviders.sessionProvider;
+  @override
+  void didChangeAppLifecycleState(AppLifecycleState state) {
+    if (state == AppLifecycleState.resumed && !_intercepting) {
+      setState(() => _loggingIn = false);
+    }
+  }
+
+  @override
+  void initState() {
+    super.initState();
+
+    WidgetsBinding.instance.addObserver(this);
+
+    final appLinks = AppLinks();
+    appLinks.uriLinkStream.listen((uri) async {
+      Logger().d('AppLinks intercepted: $uri');
+      await closeInAppWebView();
+      setState(() {
+        _intercepting = true;
+        _loggingIn = true;
+      });
+
+      if (uri.host == 'auth') {
+        WidgetsBinding.instance.addPostFrameCallback((_) async {
+          final sessionProvider =
+              Provider.of<SessionProvider>(context, listen: false);
+          try {
+            await sessionProvider.finishFederatedAuthentication(uri);
+            if (mounted) {
+              await Navigator.pushReplacementNamed(
+                context,
+                '/${NavigationItem.navPersonalArea.route}',
+              );
+            }
+          } catch (err) {
+            Logger().e('Failed to login with FederatedLogin: $err');
+          }
+        });
+      }
+
+      setState(() {
+        _loggingIn = true;
+        _intercepting = false;
+      });
+    });
+  }
+
+  @override
+  void dispose() {
+    // TODO(thePeras): Fix error used after being disposed
```

Comment:
Is this TODO going to be fixed in this PR?

Reply:
later in other 😅